tools/TestDBGenerator/Program.cs

```diff
@@ -0,0 +1,110 @@
+
+        string dbName = $"DurableDB-v{schemaVersion}";
+
+        Database db = new(dbServer, dbName);
+        db.Create();
+
+        builder.InitialCatalog = db.Name;
+
+        SqlOrchestrationServiceSettings settings = new(builder.ToString());
+        SqlOrchestrationService service = new(settings);
+        await service.CreateAsync();
+
+        Console.WriteLine($"Created database '{db.Name}'.");
+
+        Console.WriteLine($"Generating runtime data...");
+        await Orchestrations.GenerateRuntimeDataAsync(service);
+
+        string backupLocation = Path.Join(Environment.CurrentDirectory, $"{db.Name}.bak");
+        Backup backup = new()
+        {
+            Database = db.Name,
+            Action = BackupActionType.Database,
+            CopyOnly = true,
+            Incremental = false,
+            SkipTapeHeader = true,
+            UnloadTapeAfter = false,
+            NoRewind = true,
+            FormatMedia = true,
+            Initialize = true,
+            Devices =
+            {
+                new BackupDeviceItem(backupLocation, DeviceType.File),
+            },
+        };
+
+        Console.WriteLine($"Backing up database to disk...");
+        backup.SqlBackup(dbServer);
+
+        Console.WriteLine($"Created backup file '{backupLocation}'.");
+
+        // Drop the original database so that it can be restored
+        db.UserAccess = DatabaseUserAccess.Restricted;
+        db.Alter(TerminationClause.RollbackTransactionsImmediately);
+        db.Refresh();
+        db.Drop();
+
+        // Restore
+        Restore restore = new()
+        {
+            Database = $"{db.Name}-restored",
+            Devices =
+            {
+                new BackupDeviceItem(backupLocation, DeviceType.File),
+            },
+        };
+
+        Console.WriteLine("Restoring database from file (for validation)...");
+        restore.SqlRestore(dbServer);
+
+        Console.WriteLine($"Restored backup as '{db.Name}-restored");
+
+        Console.WriteLine("Compressing backup file...");
+
+        // Save to a zip file, which produces around 90% compression
+        string zipFilePath = backupLocation + ".zip";
+        using (ZipArchive archive = ZipFile.Open(zipFilePath, ZipArchiveMode.Create))
+        {
+            archive.CreateEntryFromFile(backupLocation, Path.GetFileName(backupLocation));
+        }
+
+        Console.WriteLine("Generated zipped backup file: " + zipFilePath);
+
+        // Delete the uncompressed file
+        File.Delete(backupLocation);
+    }
+
+    static string GetConnectionString()
+    {
+        string? connectionString = Environment.GetEnvironmentVariable("SQLDB_Connection");
+        if (!string.IsNullOrEmpty(connectionString))
+        {
+            return connectionString;
+        }
+
+        Console.Error.WriteLine("Specify the database connection string in the 'SQLDB_Connection' environment variable.");
+        Environment.Exit(1);
+        return string.Empty;
```

Comment:
nitpick: do we need the `return` statement here? It appears to me that `Environm.Exit()` would immediately terminate the program.

Reply:
Yes, `return` is needed in order for the program to compile. It obviously should never run, but unlike `throw` statements, the C# compiler doesn't understand that no code will execute after  `Environment.Exit()`.